### PR TITLE
Bit 553 bug fixes

### PR DESCRIPTION
**Pull request by isabella618033**
- axon priority_threadpool not passed through
- remove server's error message on validator side
- server set weight block time to be independent with optimisation step
- axon wasnt timing out at 12 sec
-  responsive uid not counted properly

### CODE REVIEW

File: bittensor/_threadpool/priority_thread_pool_impl.py [hunk 1]
```diff
@@ -190,6 +193,7 @@ def weakref_cb(_, q=self._work_queue):
         if num_threads < self._max_workers:
             thread_name = '%s_%d' % (self._thread_name_prefix or self,
                                      num_threads)
+            print('thread_name', self._max_workers, thread_name)
```
Comment (Eugene-hu) [inline bittensor/_threadpool/priority_thread_pool_impl.py]: Might want to remove this ;)

File: bittensor/_threadpool/priority_thread_pool_impl.py [hunk 2]
```diff
@@ -166,11 +166,14 @@ def submit(self, fn, *args, **kwargs):
             start_time = time.time()
             if 'priority' in kwargs:
                 del kwargs['priority']
+            
+            timeout = kwargs.get('timeout', None)
+            if 'timeout' in kwargs:
```
Comment (Eugene-hu) [inline bittensor/_threadpool/priority_thread_pool_impl.py]: Can you add tests for this individually? There should be a test_priority_thread_pool.py

File: bittensor/_axon/axon_impl.py [hunk 3]
```diff
@@ -280,7 +281,8 @@ def finalize_codes_stats_and_logs():
                     inputs_x = deserialized_forward_tensors, 
                     synapses = synapses,
                     priority = priority,
-                    hotkey= request.hotkey
+                    hotkey = request.hotkey,
+                    timeout = self.forward_timeout - (clock.time() - start_time)
```
Comment (Eugene-hu) [inline bittensor/_axon/axon_impl.py]: One thing here; we will need a higher timeout for seq2seq requests. So we need to figure out which synapse type and set the corresponding timeout

**Review by Eugene-hu — COMMENTED** [review]
Looks good! Left a few comments that we need to be careful about

File: bittensor/_axon/axon_impl.py (hunk 3 above)
Comment (isabella618033) [reply bittensor/_axon/axon_impl.py]: i think one of the issue is that it is possible for the validator (or client) to send through multiple synapse at the same time, and the way we handle it is to send multiple requests back at one time
so the timeout should be the min of the timeouts if there is multiple synapses, what do you think?

Comment (Eugene-hu) [reply bittensor/_axon/axon_impl.py]: yeap, that sounds good

**Review by Eugene-hu — APPROVED** [review]
LGTM!

**Pull request merged**